elements/auras.lua

```diff
@@ -146,7 +146,7 @@ local function createAuraIcon(element, index)
 	return button
 end
 
-local function customFilter(element, unit, button, name)
+local function customFilter(element, _, button, name)
```

Comment:
I'm not a fan of this one, since it "hides" the reference implementation signature (although it _is_ documented further below so it's probably fine).

Reply:
I could ignore unused arguments altogether:
```diff
ignore = {
-	'2/self', -- unused argument self
-	'2/event', -- unused argument event
+	'212', -- unused arguments
	'3/event', -- unused value event
	'4', -- shadowing
	'631', -- line is too long
}
```

Reply:
Or we could ignore unused arguments for elements only:
```
files["**/elements/*.lua"].ignore = {
	'212', -- unused arguments
}
```

Reply:
We could also use `--luacheck: no unused` at the end of the line to turn it off just there. Those are all options I think. I find relying on the docs alright though, but I understand what you mean.

Reply:
I agree with @p3lim on this one. In all honestly, unused arguments aren't really an issue, if there's an unused arg, in most cases it's there for a reason, for instance, like what p3lim described. Alternatively, someone messed while implementing stuff. So, imho, we should highlight unused args with the exception of unused `self`. Also, inlining is kaka, don't do it.

Reply:
We decided to use `--luacheck: no unused` where it makes sense to keep the documentation value if the argument names.

Reply:
We decided to use more specific exceptions (212, 312) for arguments named `event` and `unit`.